news/_posts/2020-10-09-how-to-write-a-new-ui-test.md

```diff
@@ -0,0 +1,407 @@
+  await ordersPage.filterOrders(
+    page, // Browser tab
+    'input', // Filter type (input or select)
+    'customer', // Filter column
+    DefaultAccount.lastName, // Filter value 'DOE'
+  );
+});
+
+it('should check customer link', async function () {
+  // Action
+  // Click on customer link in first row
+  await ordersPage.viewCustomer(
+    page, // Browser tab
+    1, // First row in list
+  );
+});
+```
+Here we use different **actions** methods, like `goToSubMenu()` in the dashboard page, or `resetAndGetNumberOfLines()` in the orders page.
+
+Note: To filter orders, the method uses the FO default account which is part of demo data stored in *data* directory. For this scenario, you need to require this file:
+
+```js
+// Import customer 'J. DOE'
+const {DefaultAccount} = require('@data/demo/customer');
+```
+
+### Expected results
+
+Data reporting (or "expected results") is done by checking elements in the page and returning a value (number of lines in a table, content of a modal, value of an input, etc).
+
+This framework uses the *expect* element from the [Chai library](https://www.chaijs.com/), so we need to require it:
+
+```js
+// Import expect from chai
+const {expect} = require('chai');
+```
+
+You can then add your `expect` calls in the test logic:
+
+```js
+it('should go to orders page', async function () {
+  // Action
+  await dashboardPage.goToSubMenu(
+    page,
+    dashboardPage.ordersParentLink,
+    dashboardPage.ordersLink,
+  );
+
+  // Expected result
+  // Verify that the current page is orders by checking the title
+  const pageTitle = await ordersPage.getPageTitle(page);
+  await expect(pageTitle).to.contains(ordersPage.pageTitle); //we compare with the value stored in the page object
+});
+
+it('should reset all filters', async function () {
+  // Action
+  const numberOfOrders = await ordersPage.resetAndGetNumberOfLines(page);
+
+  // Expected result
+  // Check that number of orders > 0 (that there's at least one element in the list) 
+  await expect(numberOfOrders).to.be.above(0);
+});
+
+it('should filter order by customer name', async function () {
+  // Action
+  await ordersPage.filterOrders(
+    page,
+    'input',
+    'customer',
+    DefaultAccount.lastName,
+  );
+
+  // Expected result
+  // Check that we have at least 1 order for the customer named "J. DOE"
+  const numberOfOrders = await ordersPage.getNumberOfElementInGrid(page);
+  await expect(numberOfOrders).to.be.at.least(1);
+});
+
+it('should check customer link', async function () {
+  // Action
+  page = await ordersPage.viewCustomer(page, 1);
+
+  // Expected result
+  // Verify that the current page is "view customer" by checking the title 'View information about J. DOE'
+  const pageTitle = await viweCustomerPage.getPageTitle(page);
+  await expect(pageTitle).to
+    .contains(`${viweCustomerPage.pageTitle} ${DefaultAccount.firstName[0]}. ${DefaultAccount.lastName}`);
```

Comment:
```suggestion
    .contains(`${viewCustomerPage.pageTitle} ${DefaultAccount.firstName[0]}. ${DefaultAccount.lastName}`);
```

You should check if the typo exists in the original file.

Reply:
fixed here and i will check on the original one

Reply:
It was wrong but in another way 
https://github.com/PrestaShop/PrestaShop/pull/21452